emukit/core/acquisition/integrated_acquisition.py

```diff
@@ -1,6 +1,7 @@
 from typing import Union, Callable, Tuple
 
 import numpy as np
+import GPy
```

Comment:
I am afraid that's a big "no". Emukit core (bayes opt, exp design, quadrature, parameter space, optimization loop) cannot assume, or depend on, specific backends. Moreover, I don't see why is that needed in this case. We already have an interface that model can imlement to generate samples of hyper-parameters, and we just need to call it here.

Perhaps I am missing something here, some good reason why we cannot make a model method call. In that case we can discuss alternative approaches.

Reply:
OK...in a narrow sense I can easily remove the dependency by removing the "import GPy" and the "if not isinstance(gpy_model, GPy.core.Model): return" - the former is just to enable the latter, and the latter is just for safety. (And I would also rename "gpy_model" to "model").

If that's not enough, I can reinstate generate_hyperparameters_samples in gpy_model_wrappers.py. I made that change simply because the call from update_parameters to generate_hyperparameters_samples involved passing a lot of attributes of IntegratedHyperParameterAcquisition into GPyModelWrapper, and the only attribute of GPyModelWrapper used inside generate_hyperparameters_samples was "model". So that piece of code seemed to me most naturally to belong in IntegratedHyperParameterAcquisition. But I don't have a good understanding of Emukit, I've just taken on the task of getting these changes merged, so I'm happy to follow guidance. Just let me know which of these options you'd like.

Reply:
No worries, and thanks for bearing with my meticulous comments. I believe it was reasonable to have it in the wrapper, so I would suggest to keep it there, stateless as it was.

Reply:
Done (the latter option).